packages/cli/internal/pkg/cli/account_activate.go

```diff
@@ -61,62 +65,97 @@ func newAccountActivateOpts(vars accountActivateVars) (*accountActivateOpts, err
 		stsClient:           aws.StsClient(profile),
 		s3Client:            aws.S3Client(profile),
 		cdkClient:           cdk.NewClient(profile),
+		cfnClient:           aws.CfnClient(profile),
 		region:              aws.Region(profile),
 	}, nil
 }
 
 // Execute activates AGC.
 func (o *accountActivateOpts) Execute() error {
+
+	environmentVars, err := o.generateEnvVars()
+	if err != nil {
+		return err
+	}
+
+	homeDir, err := osutils.DetermineHomeDir()
+	if err != nil {
+		return err
+	}
+
+	cdkAppPath := filepath.Join(homeDir, cdkCoreDir)
+	err = o.cdkBootstrap(cdkAppPath, environmentVars)
+	if err != nil {
+		return err
+	}
+
+	return o.deployCoreInfrastructure(cdkAppPath, environmentVars)
+}
+
+func (o accountActivateOpts) generateDefaultBucket() (string, error) {
+	account, err := o.stsClient.GetAccount()
+	if err != nil {
+		return "", err
+	}
+	return generateBucketName(account, o.region), nil
+}
+
+func (o accountActivateOpts) generateEnvVars() ([]string, error) {
 	if o.bucketName == "" {
 		bucketName, err := o.generateDefaultBucket()
 		if err != nil {
-			return err
+			return nil, err
 		}
 		o.bucketName = bucketName
 	}
 
 	exists, err := o.s3Client.BucketExists(o.bucketName)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	environmentVars := []string{
-		fmt.Sprintf("%s=%s", constants.AgcBucketNameEnvKey, o.bucketName),
-		fmt.Sprintf("%s=%t", constants.CreateBucketEnvKey, !exists),
-		fmt.Sprintf("%s=%s", constants.AgcVersionEnvKey, version.Version),
-	}
-	if o.vpcId != "" {
-		environmentVars = append(environmentVars, fmt.Sprintf("%s=%s", constants.VpcIdEnvKey, o.vpcId))
+		fmt.Sprintf("AGC_BUCKET_NAME=%s", o.bucketName),
+		fmt.Sprintf("CREATE_AGC_BUCKET=%t", !exists),
+		fmt.Sprintf("AGC_VERSION=%s", version.Version),
 	}
 
 	if o.customTags != nil {
 		jsonBytes, err := json.Marshal(o.customTags)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		environmentVars = append(environmentVars, fmt.Sprintf("%s=%s", constants.CustomTagEnvKey, string(jsonBytes)))
 	}
 
-	homeDir, err := osutils.DetermineHomeDir()
+	vpcId, err := o.getVpcId()
 	if err != nil {
-		return err
+		return nil, err
 	}
-
-	cdkAppPath := filepath.Join(homeDir, cdkCoreDir)
-	err = o.cdkBootstrap(cdkAppPath, environmentVars)
-	if err != nil {
-		return err
+	if vpcId != "" {
+		environmentVars = append(environmentVars, fmt.Sprintf("VPC_ID=%s", vpcId))
 	}
-
-	return o.deployCoreInfrastructure(cdkAppPath, environmentVars)
+	return environmentVars, err
 }
 
-func (o accountActivateOpts) generateDefaultBucket() (string, error) {
-	account, err := o.stsClient.GetAccount()
-	if err != nil {
-		return "", err
+func (o accountActivateOpts) getVpcId() (string, error) {
+	if o.vpcId != "" {
+		return o.vpcId, nil
+	} else {
+		stackOutputs, err := o.cfnClient.GetStackOutputs(coreStackName)
+		if err != nil {
+			if err == cfn.StackDoesNotExistError {
```

Comment:
Go error handling can be weird. I think what you have works but can you use `errors.Is(err, cfn.StackDoesNotExistError)` instead?